### PR TITLE
Add explanations and thumbnail images for layers in Style Spec docs

### PR DESCRIPTION
**Pull request by domlet**
Closes https://github.com/mapbox/mapbox-gl-js/issues/4123

Grabs [layer descriptions](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#type) from the same page:

```
"fill": A filled polygon with an optional stroked border.
"line": A stroked line.
"symbol": An icon or a text label.
"circle": A filled circle.
"heatmap": A heatmap.
"fill-extrusion": An extruded (3D) polygon.
"raster": Raster map textures such as satellite imagery.
"hillshade": Client-side hillshading visualization based on DEM data. Currently, the implementation only supports Mapbox Terrain RGB and Mapzen Terrarium tiles.
"background": The background color or pattern of the map.
```

And replicates them in each layer section – **click the arrow** for expanded screenshot:

<details>

FireShot Capture 126 - Layers - Style Specification - Mapbox - localhost

</details>

Tagging @asheemmamoowala @katydecorah for review 🙏

### CODE REVIEW

**Comment by katydecorah** [comment]
@domlet A few suggestions:

1. I think we could improve the introduction statement for each layer by answering: what the layer is and why/when to use it. From [Minh's comment]( https://github.com/mapbox/mapbox-gl-js/issues/4123#issuecomment-276814352) there is some great narrative in mapbox-gl-native that we could incorporate: https://github.com/mapbox/mapbox-gl-native/blob/01b557aeb34d51b2a63ee54ef38192f705d068e9/platform/darwin/scripts/style-spec-overrides-v8.json#L5-L22.
2.I think we should help describe how each image is styled to help answer: how can I do that? We can link out to blog posts, tutorials, or guides for the reader to learn more about the given style.
3. What do you think about using fill-width images to fill the page a little more?

Here's an example of those ideas in practice:

> image

**Comment by domlet** [comment]
Gorgeous. Thanks @katydecorah – I'll enhance the content following ^ your example.

**Comment by domlet** [comment]
Graduating this back from Draft to open for review. 😁 

Here's a preview of the changes to `layers.md` content:

<a href="https://github.com/mapbox/mapbox-gl-js-docs/files/5461641/layers.pdf"></a>

**Comment by domlet** [comment]
Thanks @ryanhamley for your review of this new content for 9 layer types. 

I'll confer with @katydecorah on options for moving the image 'captions' below the images, and seeing if we can format them in a way that clearly conveys that the captions refer specifically to the images provided – if we could use something like the [**Card**](https://mapbox.github.io/dr-ui/#card) dr-ui component (without the level/language indicators) that would work for me.

File: docs/pages/style-spec/layers.md [hunk 1]
```diff
@@ -55,38 +57,137 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
 ## line
 
+A `line` style layer renders one or more stroked polylines on the map. You can use a line layer to configure the visual appearance of polyline or multipolyline features.
+
+This map of a [Strava](https://www.mapbox.com/showcase/strava) user's hike through Grand Teton National Park uses the [`line-color`](/mapbox-gl-js/style-spec/layers/#paint-line-line-color) and [`line-width`](/mapbox-gl-js/style-spec/layers/#paint-line-line-width) paint properties to style the strong red line of the user's route.
+
+{{
+  <AppropriateImage
+    imageId="layer-line"
+    alt="Outdoors style map with a dotted red line showing a hiking path."
```
Comment (ryanhamley) [inline docs/pages/style-spec/layers.md:97]: The line isn't dotted

File: docs/pages/style-spec/layers.md [hunk 2]
```diff
@@ -55,38 +57,137 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
+  />
+}}
+
 {{<Items headingLevel="3" entry={combineItems(['layout','paint'], 'raster')} /> }}
 
 ## circle
 
+A `circle` style layer renders one or more filled circles on a map. You can use a circle layer to configure the visual appearance of point or point collection features in vector tiles. A circle layer renders circles whose radii are measured in screen units.
+
+This [cluster map](/mapbox-gl-js/example/cluster/) uses a circle layer with a GeoJSON data source, and sets the source's [`cluster`](mapbox-gl-js/style-spec/sources/#geojson-cluster) property to `true` to visualize points as clusters.
```
Comment (ryanhamley) [inline docs/pages/style-spec/layers.md:137]: I think the comma after `GeoJSON data source` isn't necessary

File: docs/pages/style-spec/layers.md [hunk 3]
```diff
@@ -55,38 +57,137 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
 ## fill-extrusion
 
+A `fill-extrusion` style layer renders one or more filled (and optionally stroked) extruded (3D) polygons on a map. You can use a fill-extrusion layer to configure the extrusion and visual appearance of polygon or multipolygon features.
+
+This map uses an external dataset to provide data-driven values for the [`fill-extrusion-height`](/mapbox-gl-js/style-spec/layers/#paint-fill-extrusion-fill-extrusion-height) paint property of various [country polygons](https://blog.mapbox.com/high-resolution-administrative-country-polygons-in-studio-57cf4abb0768) in a fill-extrusion layer.
+
+{{
+  <AppropriateImage
+    imageId="layer-fill-extrusion"
+    alt="Dark style map with data-driven extrusions rising up in 3D, in different shades of blue."
```
Comment (ryanhamley) [inline docs/pages/style-spec/layers.md:157]: This alt text seems to be for a different image

File: docs/pages/style-spec/layers.md [hunk 4]
```diff
@@ -55,38 +57,137 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
+  />
+}}
+
 {{<Items headingLevel="3" entry={combineItems(['layout','paint'], 'fill-extrusion')} /> }}
 
 ## heatmap
 
+A `heatmap` style layer renders a range of colors to represent the density of points in an area.
+
+[This visualization of earthquake data](/mapbox-gl-js/example/heatmap-layer/) uses a heatmap layer with carefully defined [paint](/mapbox-gl-js/style-spec/layers/#paint-property) properties to highlight areas where earthquake frequency is high, and many points are clustered closely together.
```
Comment (ryanhamley) [inline docs/pages/style-spec/layers.md:167]: I think the comma after `high` isn't necessary

**Review by ryanhamley — CHANGES_REQUESTED** [review]
I think this looks great! I made a few nitpick comments on wording and grammar and a couple of the alt texts appear to be outdated.

File: docs/pages/style-spec/layers.md [hunk 5]
```diff
@@ -55,38 +57,137 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
 ## fill
 
+A `fill` style layer renders one or more filled (and optionally stroked) polygons on a map. You can use a fill layer to configure the visual appearance of polygon or multipolygon features.
+
+This map of Washington, D.C. uses the [`fill-opacity`](/mapbox-gl-js/style-spec/layers/#paint-fill-fill-opacity) paint property to render a semi-transparent polygon [generated by the Isochrone API](/help/tutorials/get-started-isochrone-api/#final-product), showing how far a person can walk from the center of the city in ten minutes.
+
+{{
+  <AppropriateImage
+    imageId="layer-fill"
+    alt="Map with a pink isochrone polygon surrounding the city of Atlanta."
```
[the remark this answers is not in the document]
Comment (domlet) [reply docs/pages/style-spec/layers.md]: Good catch! I've incorporated this edit: https://github.com/mapbox/mapbox-gl-js-docs/pull/440/commits/ec314ba82c5d09a4c37d892538632c3ae95200bd

File: docs/pages/style-spec/layers.md [hunk 6]
```diff
@@ -55,38 +57,137 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
 ## background
 
+The `background` style layer covers the entire map. Use a background style layer to configure a color or pattern to show below all other map content. If the background layer is transparent or omitted from the style, any part of the map view that does not show another style layer is transparent.
+
+The [Vintage map style](https://blog.mapbox.com/designing-the-vintage-style-in-mapbox-studio-9da4aa2a627f) uses a custom SVG [`background-pattern`](/mapbox-gl-js/style-spec/layers/#paint-background-background-pattern) to achieve a textured vintage look.
+
+{{
+  <AppropriateImage
+    imageId="layer-background"
+    alt="Antique map style with a brown halftone background pattern."
```
[the remark this answers is not in the document]
Comment (domlet) [reply docs/pages/style-spec/layers.md]: Good catch! I've incorporated this edit: https://github.com/mapbox/mapbox-gl-js-docs/pull/440/commits/ec314ba82c5d09a4c37d892538632c3ae95200bd

File: docs/pages/style-spec/layers.md (hunk 1 above)
Comment (domlet) [reply docs/pages/style-spec/layers.md]: Good catch! I've incorporated this edit: https://github.com/mapbox/mapbox-gl-js-docs/pull/440/commits/ec314ba82c5d09a4c37d892538632c3ae95200bd

File: docs/pages/style-spec/layers.md (hunk 2 above)
Comment (domlet) [reply docs/pages/style-spec/layers.md]: Good catch! I've incorporated this edit: https://github.com/mapbox/mapbox-gl-js-docs/pull/440/commits/ec314ba82c5d09a4c37d892538632c3ae95200bd

File: docs/pages/style-spec/layers.md (hunk 4 above)
Comment (domlet) [reply docs/pages/style-spec/layers.md]: Good catch! I've incorporated this edit: https://github.com/mapbox/mapbox-gl-js-docs/pull/440/commits/ec314ba82c5d09a4c37d892538632c3ae95200bd

File: docs/pages/style-spec/layers.md (hunk 3 above)
Comment (domlet) [reply docs/pages/style-spec/layers.md]: Good catch! I've incorporated this edit: https://github.com/mapbox/mapbox-gl-js-docs/pull/440/commits/ec314ba82c5d09a4c37d892538632c3ae95200bd

**Review by ryanhamley — APPROVED** [review]

**Comment by domlet** [comment]
Added a `caption` component with @katydecorah to this repo to make caption text distinguishable from layer explanation text:

File: docs/pages/style-spec/layers.md [hunk 7]
```diff
@@ -55,38 +57,155 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
+
+{{
+  <AppropriateImage
+    imageId="layer-circle"
+    alt="Map with circles of different sizes and colors."
+  />
+}}
+
+{{<Caption>}}
+This [cluster map](/mapbox-gl-js/example/cluster/) uses a circle layer with a GeoJSON data source and sets the source's [`cluster`](mapbox-gl-js/style-spec/sources/#geojson-cluster) property to `true` to visualize points as clusters.
```
Comment (katydecorah) [inline docs/pages/style-spec/layers.md:155]:
```suggestion
This [cluster map](/mapbox-gl-js/example/cluster/) uses a circle layer with a GeoJSON data source and sets the source's [`cluster`](/mapbox-gl-js/style-spec/sources/#geojson-cluster) property to `true` to visualize points as clusters.
```

**Review by katydecorah — APPROVED** [review]
@domlet Nice work! I just found one tiny fix and this is ready to merge!

**Pull request merged**